data-pipeline-flink/user-cache-updater-2.0/src/main/scala/org/sunbird/dp/usercache/util/UserMetadataUpdater.scala

```diff
@@ -0,0 +1,251 @@
+      .where(QueryBuilder.in("id", userOrgIds))
+      .and(QueryBuilder.eq("isrootorg", false)).allowFiltering().toString
+    val organisationInfo = cassandraConnect.findOne(organisationQuery)
+    if (null != organisationInfo) {
+      val columnDefinitions = organisationInfo.getColumnDefinitions()
+      val columnCount = columnDefinitions.size
+      for (i <- 0 until columnCount) {
+        result.put(columnDefinitions.getName(i), organisationInfo.getObject(i))
+      }
+      if (result.contains("orgname"))
+        result.put("schoolname", result.remove("orgname").getOrElse(""))
+      if (result.contains("orgcode"))
+        result.put("schooludisecode", result.remove("orgcode").getOrElse(""))
+    }
+    result
+  }
+
+  def getLocationInformation(organisationInfo: mutable.Map[String, AnyRef], metrics: Metrics, userId: String): mutable.Map[String, AnyRef] = {
+
+    val locationIds = organisationInfo.get("locationids").getOrElse(new util.ArrayList()).asInstanceOf[util.ArrayList[String]]
+    val locationInfoMap:mutable.Map[String, AnyRef] = extractLocationMetaData(readFromCassandra(
+      keyspace = config.keySpace,
+      table = config.locationTable,
+      clause = QueryBuilder.in("id", locationIds),
+      metrics
+    ))
+    locationInfoMap
+  }
+
+  def extractLocationMetaData(locationDetails: util.List[Row]): mutable.Map[String, AnyRef] = {
+    val result: mutable.Map[String, AnyRef] = mutable.Map[String, AnyRef]()
+    locationDetails.forEach((record: Row) => {
+      record.getString("type").toLowerCase match {
+        case config.stateKey => result.put(config.stateKey, record.getString("name"))
+        case config.districtKey => result.put(config.districtKey, record.getString("name"))
+        case config.blockKey => result.put(config.blockKey, record.getString("name"))
+      }
+    })
+    result
+  }
+
+  def getUserOrgId(metrics: Metrics, userId: String): util.List[String] = {
+    val userOrgIdList:util.List[String] = new util.ArrayList()
+    val userOrgIdQuery = QueryBuilder.select("organisationid").from(config.keySpace ,config.userOrgTable).where(QueryBuilder.eq("userid", userId)).toString
+    val userOrgId = cassandraConnect.find(userOrgIdQuery)
+    userOrgId.forEach(row => userOrgIdList.add(row.getString(0)))
+    userOrgIdList
+  }
+
+  def getExternalId(userMap: mutable.Map[String, AnyRef], userid: String): String = {
+    val userChannel = userMap.getOrElse("channel", "").asInstanceOf[String]
+    val externalIdQuery = QueryBuilder.select("externalid").from(config.keySpace, config.userExternalIdTable)
+      .where(QueryBuilder.eq("idtype", userChannel))
+      .and(QueryBuilder.eq("provider", userChannel))
+      .and(QueryBuilder.eq("userid", userid)).toString
+    cassandraConnect.findOne(externalIdQuery).getString("externalid")
+  }
+
+  def closeCache(): Unit = {
+    dataCache.close()
+  }
```

Comment:
Where is the cassandra connection closed?

Reply:
Done in close()